com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -53,63 +53,26 @@ public StepPhase executeAsyncStep(ExecutionWrapper execution) throws FileStorage
-        try {
-            attemptToUploadApp(execution, app, client);
-        } catch (SLException | FileStorageException e) {
-            getStepLogger().error(e, Messages.ERROR_UPLOADING_APP, app.getName());
-            logException(execution.getContext(), e);
-            throw new SLException(e, e.getMessage());
-        } catch (CloudFoundryException cfe) {
-            CloudControllerException e = new CloudControllerException(cfe);
-            getStepLogger().error(e, Messages.ERROR_UPLOADING_APP, app.getName());
-            logException(execution.getContext(), e);
-            throw e;
-        } catch (Throwable e) {
-            Throwable eWithMessage = getWithProperMessage(e);
-            logException(execution.getContext(), eWithMessage);
-            if (e instanceof Exception) {
-                // only wrap Runtime & checked exceptions as Runtime ones
-                throw new RuntimeException(eWithMessage.getMessage(), eWithMessage);
-            } else {
-                // Errors and other should be handled elsewhere
-                throw e;
-            }
-        }
-        getStepLogger().trace("Upload app step runnable for process \"{0}\" finished", execution.getContext()
-            .getProcessInstanceId());
-    }
-
-    private void attemptToUploadApp(ExecutionWrapper execution, CloudApplicationExtended app, CloudFoundryOperations client)
-        throws FileStorageException {
-        AsyncExecutionState status = AsyncExecutionState.ERROR;
-        try {
             String appArchiveId = StepsUtil.getRequiredStringParameter(execution.getContext(), Constants.PARAM_APP_ARCHIVE_ID);
             String fileName = StepsUtil.getModuleFileName(execution.getContext(), app.getModuleName());
-            getStepLogger().debug("Uploading file \"{0}\" for application \"{1}\"", fileName, app.getName());
+            
+            getStepLogger().debug(Messages.UPLOADING_FILE_0_FOR_APP_1, fileName, app.getName());
             String uploadToken = asyncUploadFiles(execution, client, app, appArchiveId, fileName);
+            getStepLogger().debug(Messages.STARTED_ASYNC_UPLOAD_OF_APP_0, app.getName());
             execution.getContext()
                 .setVariable(Constants.VAR_UPLOAD_TOKEN, uploadToken);
-            getStepLogger().debug("Started async upload of application \"{0}\"", fileName, app.getName());
-            status = AsyncExecutionState.RUNNING;
-        } finally {
-            execution.getContext()
-                .setVariable(Constants.VAR_UPLOAD_STATE, status.name());
+        } catch (CloudFoundryException cfe) {
+            CloudControllerException e = new CloudControllerException(cfe);
+            getStepLogger().error(e, Messages.ERROR_UPLOADING_APP, app.getName());
+            logException(execution.getContext(), e);
+            throw e;
+        } catch (SLException e) {
+            getStepLogger().error(e, Messages.ERROR_UPLOADING_APP, app.getName());
+            throw e;

```

Comment:
Why not logging the SLException using the logException method?

Reply:
logException() logs the exception in the process logs and as a progress message. getStepLogger().error() does the same. I think we were using logException(), because this logic was previously in the runnable.

Reply:
Aha, and why on 69 row, there is logException method usage? :D

Reply:
Nice catch! I must have missed it. :)